src/store/favStore.ts

```diff
@@ -0,0 +1,77 @@
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
+import { IProduct } from '@/models/Products'
+
+export interface FavStoreState {
+  favouriteIds: string[]
+  addFavourite: (id: string) => void
+  removeFavourite: (id: string) => void
+  getFavouriteProducts: () => Promise<IProduct[]>
+  favourites: IProduct[]
+  count: number
+}
+
+const fetchProductsByIds = async (
+  ids: string[],
+): Promise<IProduct[] | null> => {
+  try {
+    const responses = await Promise.all(
+      ids.map((id) =>
+        fetch(`${process.env.NEXT_PUBLIC_API_HOST_REMOTE}/products/${id}`),
+      ),
+    )
+
+    if (!responses.every((response) => response.ok)) {
+      throw new Error('One or more products not found')
+    }
+
+    const favProducts: IProduct[] = await Promise.all(
+      responses.map((response) => response.json()),
+    )
+
+    return favProducts
+  } catch (error) {
+    console.error('Fetching products failed:', error)
+
+    return null
+  }
+}
```

Comment:
Why u use /products/${id} endpoint, rather than use products/ids ( and send full array of id's)

Reply:
fixed